### ISSUE TITLE
How to connect to the same session?

### ISSUE THREAD

**Issue by jagerzhang** [neutrinolabs/xrdp#2482]
Requirement description
User A login to the server through the "A-account" account, and hopes that user B can also see the real-time desktop of user A after login to the "A-account" account through xrdp.

Now I see a guide：
http://c-nergy.be/blog/?p=5305
![image](https://user-images.githubusercontent.com/9711651/209093522-da92a0f4-73b6-4bb7-9b00-b31c746f9fd8.png)

```
[Xvnc]
name=Xvnc
lib=libvnc.so
username=ask
password=ask
ip=127.0.0.1
port=ask-1
```
After setting as above, The port assigned after user A login is 5910, However, when user B logs in with an A-account and specifies port 5910, an error occurs.

The screenshot of the operation is as follows:

User A login A-accout with port -1:
![image](https://user-images.githubusercontent.com/9711651/209094812-57457fb9-3f67-4ea3-bcde-9a5ed61e8b63.png)

The port assigned after user A login is 5910
![image](https://user-images.githubusercontent.com/9711651/209094506-a857d479-773e-43f3-9a1f-1a704a69c73a.png)

Use B login A-accout with port  5910，error occurs：
![image](https://user-images.githubusercontent.com/9711651/209094948-b2344d4d-2a53-4507-a717-0eab4d4d8f68.png)
![image](https://user-images.githubusercontent.com/9711651/209095026-a9adc478-5f29-40ec-8f9f-5d25d36fc636.png)

Why is the password incorrect? Is this vnc password generated by another algorithm? 

When using vnc server, it is really necessary to set a vnc login password separately. This password is different from the account password. I don't know how to set this vnc password in xrdp?

**Comment by matt335672** [comment]
The password is separate from the VNC password and is per-session (and by implication, per-user).

Older versions of XRDP used the same VNC password as the user password. This had a number of issues relating to security, stale passwords, and password changes. The security implications were particularly disturbing - potentially long-term passwords protected with weak VNC encryption were committed to persistent storage. The current way of working limits VNC password lifetimes to a session.

You can do what you want to do by starting the VNC session independently of XRDP.  XRDP only generates a VNC password for the session if it starts that session.

**Comment by jagerzhang** [comment]
> The password is separate from the VNC password and is per-session (and by implication, per-user).
> 
> Older versions of XRDP used the same VNC password as the user password. This had a number of issues relating to security, stale passwords, and password changes. The security implications were particularly disturbing - potentially long-term passwords protected with weak VNC encryption were committed to persistent storage. The current way of working limits VNC password lifetimes to a session.
> 
> You can do what you want to do by starting the VNC session independently of XRDP. XRDP only generates a VNC password for the session if it starts that session.

@matt335672 Thanks for your reply,

How can I get this VNC password from the exist VNC session?

Or which old version VNC password is the same as the account password? We can accept this security issue

**Comment by matt335672** [comment]
@jagerzhang - the unencrypted password isn't accessible from the current session. It's not stored anywhere.

As a workaround, you can use `vncpasswd` to set the password to anything you like. You can then connect to the session by using the port method you've got above.

Also, try a version of xrdp before v0.9.x and see if it meets your requirements

**Comment by jagerzhang** [comment]
# Try 1
> As a workaround, you can use `vncpasswd` to set the password to anything you like. You can then connect to the session by using the port method you've got above.

I used `vncpasswd` to reset a new password in the new version of xrdp, and the connection also prompted a password error. It seems that the existing session password cannot be changed.

# Try 2
> Also, try a version of xrdp before v0.9.x and see if it meets your requirements

I try to compile and install v0.8 in the following way:
```
./bootstrap
./configure
make && make install
```
create service:
```
cat > /usr/lib/systemd/system/xrdp.service <<EOF
[Unit]
Description=xrdp daemon
Requires=xrdp-sesman.service
After=syslog.target network.target xrdp-sesman.service

[Service]
Type=forking
PIDFile=/var/run/xrdp.pid
EnvironmentFile=/etc/sysconfig/xrdp
ExecStart=/usr/local/sbin/xrdp $XRDP_OPTIONS
ExecStop=/usr/local/sbin/xrdp $XRDP_OPTIONS --kill

[Install]
WantedBy=multi-user.target
EOF

cat > /usr/lib/systemd/system/xrdp-sesman.service <<EOF
[Unit]
Description=xrdp session manager
After=syslog.target network.target
StopWhenUnneeded=true

[Service]
Type=forking
PIDFile=/var/run/xrdp-sesman.pid
EnvironmentFile=/etc/sysconfig/xrdp
ExecStart=/usr/local/sbin/xrdp-sesman $SESMAN_OPTIONS
ExecStop=/usr/local/sbin/xrdp-sesman $SESMAN_OPTIONS --kill

[Install]
WantedBy=multi-user.target
EOF
```

start service
```
systemctl start xrdp

netstat -nutlp | grep xrdp
tcp        0      0 127.0.0.1:3350          0.0.0.0:*               LISTEN      85766/xrdp-sesman   
tcp        0      0 0.0.0.0:3389            0.0.0.0:*               LISTEN      86773/xrdp       
```

service is succcess start, But the screen is black after I connect.

**Comment by matt335672** [comment]
Let's take these one at a time:-

For Try 1, if you crate a session using the Xvnc backend and then change the VNC password you will not be able to connect to the session using the Xvnc backend, as the passwords will differ. You can however connect directly to the VNC session by not going via sesman.

For try two, have a look in the usual log files `/var/log/xrdp.log` and `/var/log/xrdp-sesman.log` for more information on what is happening.

**Comment by jagerzhang** [comment]
It seems that the Xvnc process failed to start, Is there any operation guide for 0.8 compilation and installation？


tail -f /var/log/xrdp.*
```
==> /var/log/xrdp.log <==
[20230102-17:35:53] [INFO ] A connection received from: x.x.x.x port 51934
[20230102-17:35:53] [INFO ] An established connection closed to endpoint: x.x.x.x:51934 - socket: 8
[20230102-17:35:53] [INFO ] An established connection closed to endpoint: NULL:NULL - socket: 7
[20230102-17:35:53] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/cert.pem
[20230102-17:35:53] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/key.pem
[20230102-17:35:54] [DEBUG] xrdp_0000a708_wm_login_mode_event_00000001
[20230102-17:35:54] [WARN ] local keymap file for 0x0804 found and dosen't match built in keymap, using local keymap file
[20230102-17:35:54] [DEBUG] returnvalue from xrdp_mm_connect 0

==> /var/log/xrdp-sesman.log <==
[20230102-17:35:54] [INFO ] A connection received from: 127.0.0.1 port 46510
[20230102-17:35:54] [INFO ] scp thread on sck 10 started successfully

==> /var/log/xrdp.log <==
[20230102-17:35:54] [INFO ] xrdp_mm_process_login_response: login failed

==> /var/log/xrdp-sesman.log <==
[20230102-17:35:54] [INFO ] An established connection closed to endpoint: 127.0.0.1:46510 - socket: 10

==> /var/log/xrdp.log <==
[20230102-17:35:54] [DEBUG] xrdp_mm_module_cleanup
[20230102-17:35:54] [INFO ] An established connection closed to endpoint: 127.0.0.1:3350 - socket: 11
[20230102-17:36:26] [INFO ] A connection received from: x.x.x.x port 51952
[20230102-17:36:26] [INFO ] An established connection closed to endpoint: x.x.x.x:51952 - socket: 8
[20230102-17:36:26] [INFO ] An established connection closed to endpoint: NULL:NULL - socket: 7
[20230102-17:36:26] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/cert.pem
[20230102-17:36:26] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/key.pem
```

**Comment by matt335672** [comment]
@jagerzhang - there's nothing in the logs you've posted about Xvnc failing to start. Where did you get this information from?
